### PR TITLE
Backport of Update x/net and x/crypto/ssh into release/1.12.x

### PR DESCRIPTION
**Pull request by hc-github-team-secure-vault-core**
## Backport

This PR is auto-generated from #19282 to be assessed for backporting due to the inclusion of the label backport/1.12.x.

**WARNING** automatic cherry-pick of commits failed. Commits will require human attention.

> merge conflict error: POST https://api.github.com/repos/hashicorp/vault/merges: 409 Merge conflict []

The below text is copied from the body of the original PR.

---

I'll make a patch release for api and sdk once this is merged.

---

<details>
<summary> Overview of commits </summary>

  - 4df7c64d05a88e8c4aa3c6a5bb8791854f1f34e6 

</details>

### CODE REVIEW

**Comment by tomhjp** [comment]
Cherry picking failed, so I've rerun `go get golang.org/x/net@latest && go get golang.org/x/crypto@latest && go mod tidy` in the root, api, and sdk folders.

**Review by cipherboy — APPROVED** [review]

File: api/go.sum [hunk 1]
```diff
@@ -26,6 +26,7 @@ github.com/client9/misspell v0.3.4/go.mod h1:qj6jICC3Q7zFZvVWo7KLAzC3yx5G7kyvSDk
 github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f/go.mod h1:M8M6+tZqaGXZJjfX53e64911xZQV5JYwmTeXPW+k8Sc=
 github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
 github.com/cncf/xds/go v0.0.0-20210805033703-aa0b78936158/go.mod h1:eXthEFrGJvWHgFFCl3hGmgk+/aYT6PnTQLykKQRLhEs=
+github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
```
Comment (cipherboy) [inline api/go.sum:29]: @tomhjp Do you know where these came from? I don't see them in hte other PRs and the `go.mod` isn't enlightening me as to why these were pulled in.

Comment (tomhjp) [reply api/go.sum:29]: Good question, and indeed:

```shell-session
$ go mod why github.com/creack/pty
# github.com/creack/pty
(main module does not need package github.com/creack/pty)
```

And yet if I delete it and re-run `go mod tidy`, it still re-adds it. I've spent a while digging into this, and haven't been able to figure this out. However, I'm inclined to accept it as-is, as the `go.mod` changes still look sensible.

Comment (cipherboy) [reply api/go.sum:29]: No, this is on me. `x/crypto` is so old, apparently we didn't bump the API version of `x/crypto` in 1.12, but we did in 1.11. This is good!

Comment (tomhjp) [reply api/go.sum:29]: Ah thanks, good to have an explanation.

**Pull request merged**